src/Test/AbstractIntegrationTestCase.php

```diff
@@ -269,14 +262,27 @@ function (FixerInterface $fixer) {
                 new NullCacheManager()
             );
 
+            Tokens::clearCache();
             $runner->fix();
             $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
-            );
+            // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+            // If not, any order is valid.
+            if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+                $this->assertGreaterThan(
+                    1,
+                    count(array_unique(array_map(
+                        function (FixerInterface $fixer) {
+                            return $fixer->getPriority();
+                        },
+                        $fixers
+                    ))),
+                    sprintf(
+                        'Rules priorities are not differential enough. If rules would be used in reverse order then final output would be different than the expected one. For that, different priorities must be set up for used rules to ensure stable order of them. In "%s".',
+                        $case->getFileName()
+                    )
+                );
+            }
```

Comment:
Ok so this will only check the case where 2 (or more) fixers have the same priority and the fixer output are different.

When does that happens? I thought we were doing those kind of tests when priorities are different (or maybe I was in a specific use case).

What I said earlier was: since we are running the fixer twice (in the right and reversed order) and we know that fixers without the same priority will generate diff, why don't we use this second run to check that two fixers with the same priority generate the same fix?

I've also noticed (thanks to your PR) that if fixers are specified in an order different from their priorities, the IntegrationTest will produce wrong fix.